AzureADSample/AzureADSample/View/MainPage.xaml.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AzureADSample.Model;
+using AzureADSample.Service;
+using AzureADSample.Views;
+using Microsoft.Identity.Client;
+
+namespace AzureADSample;
+
+public partial class MainPage : ContentPage
+{
+    AuthService authService;
+    User user;
+    public MainPage()
+    {
+        InitializeComponent();
+        authService = new AuthService();
+        user = new User();
+    }
+
+    // If an existing account already exist
+    // then direct the user to the settigns page
+    protected override async void OnAppearing()
+    {
+        try
+        {
+            var accounts = await AuthService.authenticationClient.GetAccountsAsync();
+            AuthenticationResult result;
+
+            if (accounts.Count() >= 1)
+            {
+                result = await AuthService.authenticationClient
+                   .AcquireTokenSilent(Constants.Scopes, accounts.FirstOrDefault())
+                   .ExecuteAsync();
+
+                await SecureStorage.SetAsync("Token", result?.IdToken); // store token securely for later use
+                authService.GetUserClaims(result, user);
+                await DisplayAlert("Sucessful Logged in", "Existing account exist", "Ok");
+                await Shell.Current.GoToAsync($"{nameof(SettingsPage)}",
+                new Dictionary<string, object>
+                {
+                    [nameof(User)] = user
+                });
+                await DisplayAlert($"Welcome back {user.Name}", "", "Ok");
+            }
+        }
+        catch
+        {
+            // Do nothing - the user isn't logged in
+        }
```

Comment:
Was there a reason you wanted to use a try/catch here? I think you should be able to delete the try/catch clause since you're not doing anything in the catch.

Reply:
I removed the try/catch